chain/network/src/types.rs

```diff
@@ -267,6 +267,7 @@ impl From<Handshake> for network_proto::Handshake {
 #[derive(PartialEq, Eq, Clone, Debug)]
 pub enum PeerMessage {
     Handshake(Handshake),
+    HandshakeGenesisMismatch(PeerId, CryptoHash),
```

Comment:
let's actually make a more generic HandshakeMismatch(PeerId, MismatchReason)
with enum MismatchReason: Genesis(CryptoHash), ProtocolVersion(u32), 
and add if versions don't match it should return that as well.

Reply:
makes sense